niceman/distributions/conda.py

```diff
@@ -269,7 +283,7 @@ def identify_distributions(self, paths):
 
         # First, loop through all the files and identify conda paths
         for path in paths:
-            conda_path = self._get_conda_path(path)
+            conda_path = self._get_conda_env_path(path)
```

Comment:
so here if we have `paths=['/a/b', '/a/c']` where `/a` is not a conda path... I wondered if we could manage to avoid checking the same parent paths over and over again.

Reply:
> avoid checking the same parent paths over and over again

Beyond the caching that `PathRoot` does?

Not directly related, but I'm not sure I see the advantage of having the `_get_conda_env_path` and `_get_conda_root_path` since they are marked as internal and are just direct calls to the respective `PathRoot` objects.

Reply:
I have to admit that I found it easier to read.  However, it does hide the fact that PathRoot caches the results.